### ISSUE TITLE
2021-3-13

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#479]
751Move zero_grad() before optimize; done; (2021-3-6)
752Use another way to write self.model; done; torch.nn.DataParallel(module, device_ids=None, output_device=None, dim=0); (2021-3-6)
753Let’s remove globals() to check whether it works or not; not worked, globals() should be kept; (2021-3-6)
754Read through trainer.py; (2021-3-6)
755Torch.save function? Saved file has many strange characters; done; (2021-3-6)
756What if we comment self.model.train? Wtf, it can still work; (2021-3-6)
757Change loss_f to criterion ; done; (2021-3-6)
758What is nn.CrossEntropyLoss? 交叉熵：它主要刻画的是实际输出（概率）与期望输出（概率）的距离，也就是交叉熵的值越小，两个概率分布就越接近; (2021-3-6)
759What is torch.optim.SGD? 要使用 torch.optim 要先创建一个optim 对象，这个对象会一直保持当前状态或根据计算的梯度更新参数。创建optim对象时，要给它一个包含模型参数的的可迭代对象（所有的都应该是 Variable ），然后指定learning rate，weight decay等参数; optimizer.step()和 optimizer.step(closure) 所有的optim 都实现了前一种方法，第一种方法会更新所有参数，这是大多数 optim 都支持的方法，只要 损失反向传播后就可以调用此函数; done; (2021-3-6)
